### PR TITLE
Feature: Allow admin users to be assigned to entities

### PR DESCRIPTION
**Pull request by Marc-Roig**
https://github.com/strapi/strapi/assets/20578351/69447ff2-f40b-4873-970e-61620ce9801c

### What does it do?

- Assign a user in a RW activated Content Type.
- Unassign a user in a RW activated Content Type.
- It sanitizes the assignee data so it does not filter sensitive data.

### CODE REVIEW

File: packages/core/admin/ee/server/services/review-workflows/assignees.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  return {
+    /**
+     * Update the assignee of an entity
+     */
+    async updateEntityAssignee(id, model, assigneeId) {
+      if (isNil(assigneeId)) {
+        return this.deleteEntityAssignee(id, model);
+      }
+
+      const userExists = await getService('user').exists({ id: assigneeId });
```
Comment (nathan-pichon) [inline packages/core/admin/ee/server/services/review-workflows/assignees.js:18]:
```suggestion
module.exports = ({ strapi }) => {
  return {
    /**
     * Update the assignee of an entity
     */
    async updateEntityAssignee(id, model, assigneeId) {
      if (isNil(assigneeId)) {
        return this.deleteEntityAssignee(id, model);
      }

      const userExists = await getService('user', { strapi }).exists({ id: assigneeId });
```

File: packages/core/admin/ee/server/services/review-workflows/review-workflows.js [hunk 2]
```diff
@@ -35,18 +35,26 @@ async function initDefaultWorkflow({ workflowsService, stagesService, strapi })
...
-      private: false,
-      configurable: false,
-      visible: false,
-      useJoinTable: true, // We want a join table to persist data when downgrading to CE
-      type: 'relation',
-      relation: 'oneToOne',
-      target: 'admin::workflow-stage',
-    });
-    strapi.container.get('content-types').extend(contentTypeUID, setStageAttribute);
+    const setAttribute = (path, target) =>
```
Comment (jhoward1994) [inline packages/core/admin/ee/server/services/review-workflows/review-workflows.js:38]: Nice, thanks for handling this as well in this PR!

I think this comment is relevant here too - https://github.com/strapi/strapi/pull/16889#discussion_r1218198223

**Review by jhoward1994 — APPROVED** [review]
Sorry for pushing to you PR @Marc-Roig it was just a minor thing and I thought it'd be easier that way.

This looks really good, thanks!

**Comment by Marc-Roig** [comment]
no prob! I actually had that change in my machine but forgot to push it 😓 thank you !!

**Review by nathan-pichon — APPROVED** [review]

**Pull request merged**